backend/backend.go

```diff
@@ -83,6 +83,24 @@ type Local interface {
 	Context(*Operation) (*terraform.Context, state.State, error)
 }
 
+// Creator can be implemented by backends to automatically create
+// necessary resources during `init`. The backend will be fully configured
```

Comment:
When you say "resources" here are you meaning it in the sense of Terraform's first-class concept or just in the general sense of "stuff it uses"?

Just curious, since the use of this term here made me wonder about e.g. giving the backend somewhere to keep track of what it created and having an "un-init" operation that destroys the stuff later. But probably I just have the wrong idea of what sort of resources we're talking about here.